app/models/concerns/tags_substitution_concern.rb

```diff
@@ -142,6 +148,16 @@ def external_link(url)
     link_to(url, url, target: '_blank', rel: 'noopener')
   end
 
+  def external_link_with_body(body, url)
+    link_to(body, url, target: '_blank', rel: 'noopener')
+  end
+
+  def url_for_justificatif_motivation(dossier)
+    if dossier.justificatif_motivation.attached?
+      Rails.application.routes.url_helpers.url_for(dossier.justificatif_motivation)
```

Comment:
`Rails.application.routes.url_helpers` est inclus dans la classe donc tu peux directement faire `url_for`

Reply:
alors après essai, non: https://circleci.com/gh/betagouv/demarches-simplifiees.fr/10377#tests/containers/1
La raison est expliquée là: https://stackoverflow.com/questions/20660604/using-the-correct-url-for-method-in-a-rails-engine-spec
Rails.application.routes.url_helpers et ActionView::Helpers::UrlHelper sont inclus simultanémment et rentrent en conflit. Aucune des solutions préconisées sur SO ne fonctionne ici (ça fait péter d'autres trucs) donc j'ai l'impression que la seule solution pour nous est d'explicitement appeler le bon url_for